loadtest/control/simulcontroller/actions.go

```diff
@@ -1729,72 +1729,86 @@ func (c *SimulController) viewThread(u user.User) control.UserActionResponse {
 		thread = *threads[0]
 	}
 
-	postIds, hasNext, err := u.GetPostThreadWithOpts(thread.PostId, "", model.GetPostsOptions{
+	// Get the lastUpdateAt
+	opts := model.GetPostsOptions{
 		CollapsedThreads: true,
 		Direction:        "down",
-		PerPage:          25,
-	})
-	if err != nil {
-		return control.UserActionResponse{Err: control.NewUserError(err)}
-	}
-	if len(postIds) == 0 {
-		return control.UserActionResponse{Info: "viewthread: no posts available to view in thread"}
-	}
-	newestPostId := postIds[len(postIds)-1]
-	newestPost, err := u.Store().Post(newestPostId)
-	if err != nil && !errors.Is(err, memstore.ErrPostNotFound) {
-		return control.UserActionResponse{Err: control.NewUserError(err)}
+		PerPage:          60, // Matching with webapp's PER_PAGE_DEFAULT
 	}
-	var newestCreateAt int64
-	if errors.Is(err, memstore.ErrPostNotFound) {
-		newestCreateAt = thread.Post.CreateAt
-	} else {
-		newestCreateAt = newestPost.CreateAt
+	if thread.LastUpdateAt != 0 && c.serverVersion.GTE(semver.MustParse("10.8.0")) { // Use the new logic only from v10.8 onwards.
+		opts.UpdatesOnly = true
+		opts.FromUpdateAt = thread.LastUpdateAt
 	}
 
-	// scrolling between 1 and 3 times
-	numScrolls := rand.Intn(3) + 1
-	for i := 0; i < numScrolls && hasNext; i++ {
-		postIds, hasNext, err = u.GetPostThreadWithOpts(thread.PostId, "", model.GetPostsOptions{
-			CollapsedThreads: true,
-			Direction:        "down",
-			PerPage:          25,
-			FromPost:         newestPostId,
-			FromCreateAt:     newestCreateAt,
-		})
+	// We always load the entire thread the first time.
+	// For subsequent loads, we only load the next updated posts.
```

Comment:
Where do the "subsequent loads" happen?

Reply:
They will happen when `u.Store().RandomThread()` returns the same threadId again.